build_full_frontend.sh

```diff
@@ -36,30 +36,33 @@ export DEFAULTDIR
 # create folder $DEFAULTDIR/frontend/dist/app/ if no exists
 mkdir -p $DEFAULTDIR/frontend/dist/app/ || logerror "Failed to make directory $DEFAULTDIR/frontend/dist/app/"
 
-# Tracim vendors
+# Tracim frontend vendors
 cd "$DEFAULTDIR/frontend_vendors"
-./build_vendors.sh $appdev && loggood "success" || logerror "Could not build tracim_frontend_vendors"
+./build_vendors.sh $shelldev && loggood "success" || logerror "Could not build tracim_frontend_vendors"
 
-# Tracim Lib for unit tests
-# NOTE - RJ - 2020-08-20 - the absence of $dev is intentional
-log "Building unoptimized tracim_frontend_lib for tests"
-yarn workspace tracim_frontend_lib run build && loggood "success" || logerror "Could not build tracim_frontend_lib"
+# Tracim frontend Lib as standalone
+# required to support standalone build and because tracim_frontend_lib is in app's package.json
+# INFO - CH - 2021-12-29 - It is currently only used to allow apps to import frontend_lib stylus files
+# NOTE - RJ - 2020-08-20 - the absence of $yarndev is intentional
+yarn workspace tracim_frontend_lib run build:standalone && loggood "success" || logerror "Could not build tracim_frontend_lib"
 
-# Tracim Lib for the browsers
+# Tracim frontend Lib as optimized
 log "Building optimized tracim_frontend_lib"
-yarn workspace tracim_frontend_lib run buildoptimized$dev && loggood "success" || logerror "Could not build tracim_frontend_lib for Tracim"
+cd "$DEFAULTDIR/frontend_lib"
+./build_frontend_lib.sh
 
 for app in "$DEFAULTDIR"/frontend_app_*; do
 	if [ -f "$app/.disabled-app" ]; then
 		log "Skipping $app because of the existence of the .disabled-app file"
 	else
 		cd "$app" || exit 1
-		./build_*.sh $appdev || logerror "Failed building $app."
+		./build_*.sh $shelldev || logerror "Failed building $app."
```

Comment:
```suggestion
		./build_${app}.sh $shelldev || logerror "Failed building $app."
```
I think this would work (as long as the build_xxx.sh script always has the same suffix as the app folder).

Reply:
i'll give it a try

Reply:
not sure if it would work for the `collaborative_document_edition` app, maybe the name of your build script needs to be changed

Reply:
indeed, it won't work for collaborative_document_edition

I'm going to rename only the .sh or apps and will call them build_app.sh.

So there will be:
build_vendors.sh
build_frontend_lib.sh
build_frontend.sh
build_app.sh (one for each apps)

Reply:
fixed